### PR TITLE
BUG: fix corner case bugs in power_method_opnorm

### PR DESCRIPTION
**Pull request by adler-j**
Fixes some inconsistencies as reported by @niinimaki.

### CODE REVIEW

File: odl/operator/oputils.py [hunk 1]
```diff
@@ -152,33 +159,49 @@ def power_method_opnorm(op, niter, xstart=None):
...
             op(x, out=tmp)
             op.adjoint(tmp, out=x)
-            x /= x.norm()
 
-        op(x, out=tmp)
-        op.adjoint(tmp, out=x)
-        return sqrt(x.norm())
+            x_norm = x.norm()
+            if x_norm == 0:
+                raise ValueError('reached ``x=0``')
```
Comment (kohr-h) [inline odl/operator/oputils.py]: Out of curiosity: when does this occur?

Comment (adler-j) [inline odl/operator/oputils.py]: There is an test for exactly this further down. Basically when you have an operator like

```
[[0, 1],
 [0, 0]]
```

and if you start with `[1, 1]`, you then get `[1, 0]`, and then `[0, 0]`.

File: odl/operator/oputils.py [hunk 2]
```diff
@@ -152,33 +159,49 @@ def power_method_opnorm(op, niter, xstart=None):
...
 
-    tmp = op.range.element()
+    x_norm = x.norm()
+    if x_norm == 0:
+        raise ValueError('input ``x=0``')
+    x /= x_norm
+
+    niter = int(niter)
+    if niter <= 0:
+        raise ValueError('niter not non-negative')
```
Comment (kohr-h) [inline odl/operator/oputils.py]: Better style: `'`foo` must be bar, got {}'.format(baz)`. More informative, and nice to have when the input is the result of another computation.

File: odl/operator/oputils.py (hunk 1 above)
Comment (adler-j) [inline odl/operator/oputils.py]: Some reading material [here](https://en.wikipedia.org/wiki/Nilpotent_operator)

**Pull request merged**